junit-jupiter-api/src/main/java/org/junit/jupiter/api/MethodOrderer.java

```diff
@@ -172,6 +175,76 @@ private static int getOrder(MethodDescriptor descriptor) {
 		}
 	}
 
+	/**
+	 * {@code MethodOrderer} that sorts methods based on the {@link Depend @Depend}
+	 * annotation.
+	 */
+	class DependAnnotation implements MethodOrderer {
+		private static final Logger logger = LoggerFactory.getLogger(Random.class);
+
+		private static Map<String, Integer> dependencySize;
+		private static Map<String, String[]> digraph;
+
+		@Override
+		public void orderMethods(MethodOrdererContext context) {
+			digraph = context.getMethodDescriptors().stream().filter(
+				descriptor -> descriptor.findAnnotation(Depend.class).isPresent()).collect(
+					Collectors.toMap(descriptor -> descriptor.getMethod().getName(),
+						descriptor -> descriptor.findAnnotation(Depend.class).map(Depend::methods).get()));
```

Comment:
Please redesign this so that it does not look up annotations twice.

Reply:
If you cannot figure out a good way to eliminate the double lookup, please at least change the above to the following.

```java
			// @formatter:off
			digraph = context.getMethodDescriptors().stream()
					.filter(descriptor -> descriptor.isAnnotated(Depend.class))
					.collect(toMap(
							descriptor -> descriptor.getMethod().getName(),
							descriptor -> descriptor.findAnnotation(Depend.class).map(Depend::methods).get()));
			// @formatter:on

```

Reply:
I have changed as above using the isAnnotated() method. 
I can also redesign using for loop, if and add each element one by one to avoid double lookup but I am not really sure if it is much better so I just use the isAnnotated method for now. If double lookup should be avoided at all cost, please tell me and I will rewrite it that way.